contracts/Nexus.sol

```diff
@@ -291,196 +289,43 @@ contract Nexus is INexus, BaseAccount, ExecutionHelper, ModuleManager, UUPSUpgra
         UUPSUpgradeable.upgradeToAndCall(newImplementation, data);
     }
 
-    /// @dev For automatic detection that the smart account supports the nested EIP-712 workflow.
+    /// @dev For automatic detection that the smart account supports the nested EIP-712 workflow
+    /// with a specific validator.
+    ///
+    /// Temporary implementation, not following ERC-7739 interface.
+    /// See https://ethereum-magicians.org/t/erc-7739-readable-typed-signatures-for-smart-accounts/20513/2 
+    /// for discussions.
+    ///
     /// By default, it returns `bytes32(bytes4(keccak256("supportsNestedTypedDataSign()")))`,
     /// denoting support for the default behavior, as implemented in
     /// `_erc1271IsValidSignatureViaNestedEIP712`, which is called in `isValidSignature`.
     /// Future extensions should return a different non-zero `result` to denote different behavior.
     /// This method intentionally returns bytes32 to allow freedom for future extensions.
-    function supportsNestedTypedDataSign() public pure virtual returns (bytes32 result) {
-        result = bytes4(0xd620c85a);
+    function supportsNestedTypedDataSignWithValidator(address validator) public view virtual returns (bytes32) {
```

Comment:
so we pass the validator and ask the account if this validator supports 7739? 
why can't read method just sit in the validator? is there any other usage of this

Reply:
it's fine. we wont run into codesize problems now. and its implementation contract deployed once so ok